packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts

```diff
@@ -643,40 +644,51 @@ async function calculateUserPerformance(onChainProposal: ProposalAccountWithKey)
         .execute()
     })
 
-    let actors = <any>{}
-    
-    orders.forEach(o => {
-      const actor = o.actorAcct
-      if (!actors[actor]) {
-        actors[actor] = {
+    let actors = orders.reduce((current, next) => {
+      const actor = next.actorAcct
+      if (!current.get(actor)) {
+        current.set(actor, {
           tokensBought: 0,
           tokensSold: 0,
           volumeBought: 0,
           volumeSold: 0
-        }
+        })
+      }
+
+      
+      const tokenDecimals = tokens?.decimals
+      if (!tokenDecimals) {
+        // do we want to throw different type of error here?
+        // do we want to throw error at all?
+        throw new Error("invalid token decimals")
       }
 
-      const totals = actors[actor]
-      const orderAmount = PriceMath.getHumanAmount(new BN(o.filledBaseAmount), tokens?.decimals)
-      const price = o.quotePrice * orderAmount
+      const totals = current.get(actor)
+      const orderAmount = PriceMath.getHumanAmount(new BN(next.filledBaseAmount), tokens?.decimals)
+      const price = next.quotePrice * orderAmount
 
-      if (o.side === "BID") {
+      if (next.side === "BID") {
         totals.tokensBought += orderAmount
         totals.volumeBought += price
-      } else if (o.side === "ASK") {
+      } else if (next.side === "ASK") {
         totals.tokensSold += orderAmount;
         totals.volumeSold += price;
       }
-    })
 
-    Object.keys(actors).forEach(k => {
-       const actor = actors[k]
+      current.set(actor, totals)
+
+      return current
+
+    }, new Map <String, any>())
```

Comment:
Instead of Map<String, any> can we do Map<String, PerformanceTotals> where performance totals is a type containing like tokens bought and sold and volume bought and sold etc.